cpp/tests/test_state_age_function.cpp

```diff
@@ -24,241 +24,382 @@
+    ScalarType dt = 0.5;
 
-    wrapper.set_parameter(testvalue_after);
-    EXPECT_EQ(wrapper.get_parameter(), testvalue_after);
-    EXPECT_EQ(wrapper.get_support_max(dt), testvalue_after);
-    EXPECT_EQ(wrapper.get_mean(dt), testvalue_after * 0.5);
+    mio::GammaSurvivalFunction gamma(1., 2., 3.);
+    mio::StateAgeFunctionWrapper wrapper(gamma);
+
+    EXPECT_EQ(wrapper.get_parameter(), 1.);
+    EXPECT_EQ(wrapper.get_location(), 2.);
+    EXPECT_EQ(wrapper.get_scale(), 3.);
+    EXPECT_NEAR(wrapper.get_support_max(dt), 71.5, 1e-14);
+    EXPECT_NEAR(wrapper.get_mean(), 5.0, 1e-14);
+
+    wrapper.set_parameter(4.5);
+    wrapper.set_location(0.5);
+    wrapper.set_scale(1.5);
+    EXPECT_EQ(wrapper.get_parameter(), 4.5);
+    EXPECT_EQ(wrapper.get_location(), 0.5);
+    EXPECT_EQ(wrapper.get_scale(), 1.5);
+    EXPECT_EQ(wrapper.get_support_max(dt), 50);
+    EXPECT_NEAR(wrapper.get_mean(), 7.25, 1e-4);
 }
 
 TEST(TestStateAgeFunction, testComparisonOperator)
 {
     // Check that StateAgeFunctions are only considered equal if they are of the same derived
     // class and have the same parameter.
-    mio::ExponentialDecay expdecay(0.5);
-    mio::ExponentialDecay expdecay2(0.5);
-    mio::ExponentialDecay expdecay3(1.0);
+    mio::GammaSurvivalFunction gamma(0.5, 0, 1);
+    mio::GammaSurvivalFunction gamma2(0.5, 0, 1);
+    mio::GammaSurvivalFunction gamma3(2, 0, 1);
+    mio::GammaSurvivalFunction gamma4(0.5, 1.5, 1);
+    mio::GammaSurvivalFunction gamma5(0.5, 0, 2);
     mio::SmootherCosine smoothcos(0.5);
 
-    EXPECT_TRUE(expdecay == expdecay2);
-    EXPECT_FALSE(expdecay == expdecay3);
-    EXPECT_FALSE(expdecay == smoothcos);
+    EXPECT_TRUE(gamma == gamma2);
+    EXPECT_FALSE(gamma == gamma3);
+    EXPECT_FALSE(gamma == gamma4);
+    EXPECT_FALSE(gamma == gamma5);
+    EXPECT_FALSE(gamma == smoothcos);
 
-    // Check that it also holds when a StateAgeFunctionWrapper is set with the respective functions.
-    mio::StateAgeFunctionWrapper wrapper(expdecay);
-    mio::StateAgeFunctionWrapper wrapper2(expdecay2);
-    mio::StateAgeFunctionWrapper wrapper3(expdecay3);
-    mio::StateAgeFunctionWrapper wrapper4(smoothcos);
+    // Check that it also holds when a StateAgeFunctionWrapper is set with the respective functions
+    mio::StateAgeFunctionWrapper wrapper(gamma);
+    mio::StateAgeFunctionWrapper wrapper2(gamma2);
+    mio::StateAgeFunctionWrapper wrapper3(gamma3);
+    mio::StateAgeFunctionWrapper wrapper4(gamma3);
+    mio::StateAgeFunctionWrapper wrapper5(gamma3);
+    mio::StateAgeFunctionWrapper wrapper6(smoothcos);
 
     EXPECT_TRUE(wrapper == wrapper2);
     EXPECT_FALSE(wrapper == wrapper3);
     EXPECT_FALSE(wrapper == wrapper4);
-}
\ No newline at end of file
+    EXPECT_FALSE(wrapper == wrapper5);
+    EXPECT_FALSE(wrapper == wrapper6);
+}
+
+TEST(TestStateAgeFunction, testGamma)
+{
+    // Check if connection between Gammasurvivalfunction and ErlangDensity is fulfilled.
```

Comment:
```suggestion
    // Check if connection between GammaSurvivalFunction and ErlangDensity is fulfilled.
```
Can you explain more detailed what connection exactly you are testing here?

Reply:
This is not a very elegant formula but i added a comment...